packages/fetchai/skills/registration_aw1/strategy.py

```diff
@@ -43,38 +43,47 @@ def __init__(self, **kwargs: Any) -> None:
-        enforce(
-            ethereum_address != DEFAULT_ETHEREUM_ADDRESS
-            and LedgerApis.is_valid_address("ethereum", ethereum_address),
-            f"Not a valid ethereum_address: {ethereum_address}",
-        )
-        self._ethereum_address = ethereum_address
-        signature_of_fetchai_address = kwargs.pop(
-            "signature_of_fetchai_address", DEFAULT_SIGNATURE_OF_FETCHAI_ADDRESS
-        )
-        enforce(
-            signature_of_fetchai_address != DEFAULT_SIGNATURE_OF_FETCHAI_ADDRESS
-            and isinstance(signature_of_fetchai_address, str),
-            f"Not a valid signature_of_fetchai_address: {signature_of_fetchai_address}",
-        )
-        self._signature_of_fetchai_address = signature_of_fetchai_address
         developer_handle = kwargs.pop("developer_handle", DEFAULT_DEVELOPER_HANDLE)
         enforce(
             developer_handle != DEFAULT_DEVELOPER_HANDLE
             and isinstance(developer_handle, str),
             f"Not a valid developer_handle: {developer_handle}",
         )
         self._developer_handle = developer_handle
-        tweet = kwargs.pop("tweet", DEFAULT_TWEET)
-        enforce(isinstance(tweet, str), "Not a valid tweet link")
-        self._tweet = tweet
+        self._whitelist = kwargs.pop("whitelist", DEFAULT_WHITELIST)
         self._shared_storage_key = kwargs.pop(
             "shared_storage_key", DEFAULT_SHARED_STORAGE_KEY
         )
-        self._whitelist = kwargs.pop("whitelist", DEFAULT_WHITELIST)
+        self.announce_termination_key = kwargs.pop("announce_termination_key", None)
+
+        self.developer_handle_only = kwargs.pop("developer_handle_only", False)
+        if not self.developer_handle_only:
+            ethereum_address = kwargs.pop("ethereum_address", DEFAULT_ETHEREUM_ADDRESS)
+            enforce(
+                ethereum_address != DEFAULT_ETHEREUM_ADDRESS
+                and LedgerApis.is_valid_address("ethereum", ethereum_address),
+                f"Not a valid ethereum_address: {ethereum_address}",
+            )
+            self._ethereum_address = ethereum_address
+            signature_of_fetchai_address = kwargs.pop(
+                "signature_of_fetchai_address", DEFAULT_SIGNATURE_OF_FETCHAI_ADDRESS
+            )
+            enforce(
+                signature_of_fetchai_address != DEFAULT_SIGNATURE_OF_FETCHAI_ADDRESS
+                and isinstance(signature_of_fetchai_address, str),
+                f"Not a valid signature_of_fetchai_address: {signature_of_fetchai_address}",
+            )
+            self._signature_of_fetchai_address = signature_of_fetchai_address
+            tweet = kwargs.pop("tweet", DEFAULT_TWEET)
+            enforce(isinstance(tweet, str), "Not a valid tweet link")
+            self._tweet = tweet
+            self._is_ready_to_register = False
+        else:
+            self._is_ready_to_register = True
+            self._ethereum_address = "some_dummy_address"
```

Comment:
""/None is better imo

Reply:
This messes up the property type hint for `ethereum_address` below and updating that with `Optional` leads to further mypy complains elsewhere in the handler.